dlrover/python/elastic_agent/monitor/resource.py

```diff
@@ -53,6 +54,38 @@ def get_used_memory():
     return int(mem.used / 1024 / 1024)
 
 
+def get_gpu_stats():
+    """ "Get the used gpu info of the container"""
+    device_count = pynvml.nvmlDeviceGetCount()
+    gpu_stats_list = []
+    for i in range(device_count):
+        handle = pynvml.nvmlDeviceGetHandleByIndex(i)
+        memory_info = pynvml.nvmlDeviceGetMemoryInfo(handle)
+        total_memory = memory_info.total / (1024**2)
+        used_memory = memory_info.used / (1024**2)
+
+        # Get GPU utilization
+        utilization = pynvml.nvmlDeviceGetUtilizationRates(handle)
+        gpu_utilization = utilization.gpu
+
+        gpu_stats_list.append(
+            {
+                "index": i,
+                "total_memory_mb": total_memory,
+                "used_memory_mb": used_memory,
+                "gpu_utilization": gpu_utilization,
+            }
+        )
+    for info in gpu_stats_list:
+        logger.debug(
```

Comment:
We can `logger.debug(f"GPU metric {gpu_metric}")` if we use a dataclass of GPUMetric.

Reply:
Rewrited with dataclass now. Tks much for reviewing! Learned better ways from the review to organize data!